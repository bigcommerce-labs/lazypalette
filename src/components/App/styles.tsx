import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
