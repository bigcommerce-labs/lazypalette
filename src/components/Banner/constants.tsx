import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
