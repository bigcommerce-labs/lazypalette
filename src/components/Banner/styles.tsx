import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
