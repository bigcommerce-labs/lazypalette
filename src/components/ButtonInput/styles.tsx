import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
