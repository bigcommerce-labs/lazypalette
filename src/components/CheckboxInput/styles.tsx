import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
