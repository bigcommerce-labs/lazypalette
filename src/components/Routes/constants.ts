export const enum NoMatch {
    Heading = '404 Error - Page not found',
    Body = 'Uh oh, looks like the page you are looking for has moved or no longer exists.',
}
