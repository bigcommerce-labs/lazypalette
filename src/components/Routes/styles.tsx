import styled from 'styled-components';

export const Container = styled.div`
    background: ${({ theme }) => theme.colors.empty};
    border-radius: 0px;
    box-shadow: ${({ theme }) => theme.elevation.raised};
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    margin: 0.75rem 0.75rem 0 0;
    height: calc(100% - 6rem);
    width: calc(100% - 15.5rem);
`;

export const Heading = styled.h1`
    color: ${({ theme }) => theme.colors.primaryText};
`;

export const Body = styled.p`
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
`;
