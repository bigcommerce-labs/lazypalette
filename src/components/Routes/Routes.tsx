import React, { PureComponent } from 'react';
import { withRouter, Route, RouteComponentProps, Switch } from 'react-router-dom';

import PreviewPane from '../PreviewPane/PreviewPane';

import { NoMatch } from './constants';
import { Body, Container, Heading } from './styles';

export const appRoutes = {
    history: {
        path: '/history',
        route: 'history',
    },
    home: {
        path: '/',
        route: '/',
    },
    options: {
        path: '/options',
        route: 'options',
    },
    section: {
        path: '/section/:id(\\d+)',
        route: 'section/',
    },
    styles: {
        path: '/styles',
        route: 'styles',
    },
};

export const NoRoute = () => (
    <Container>
        <Heading>{NoMatch.Heading}</Heading>
        <Body>{NoMatch.Body}</Body>
    </Container>
);

interface RoutesProps extends RouteComponentProps<{}> {}

export class Routes extends PureComponent<RoutesProps> {
    render() {
        const { history, home, options, styles, section  } = appRoutes;

        return (
            <Switch>
                <Route path={history.path} exact component={PreviewPane} />
                <Route path={home.path} exact component={PreviewPane} />
                <Route path={options.path} exact component={PreviewPane} />
                <Route path={styles.path} exact component={PreviewPane} />
                <Route path={section.path} exact component={PreviewPane} />
                <Route component={NoRoute} />
            </Switch>
        );
    }
}

export default withRouter(Routes);
