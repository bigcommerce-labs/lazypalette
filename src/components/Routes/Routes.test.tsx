import { mount, shallow } from 'enzyme';
import React from 'react';
import {Provider} from 'react-redux';
import { StaticRouter } from 'react-router';
import createMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';

import PreviewPane from '../PreviewPane/PreviewPane';

import { NoMatch } from './constants';
import { Body, Heading } from './styles';
import { appRoutes, NoRoute, Routes } from './Routes';

describe('Routes', () => {
    const { home, styles } = appRoutes;
    const theme = {
        colors: {},
        elevation: {},
        layers: {},
        typography: {
            fontSize: {},
            fontWeight: {},
        },
    };
    const mockProp: any = jest.fn();
    const routeProps = {
        history: mockProp,
        location: mockProp,
        match: mockProp,
        staticContext: mockProp,
    };

    it('renders', () => {
        const routes = shallow(
            <StaticRouter location={home.path} context={{}}>
                <Routes {...routeProps} />
            </StaticRouter>
        );

        expect(routes).toMatchSnapshot();
    });

    describe('when route matches path', () => {
        it('should render the preview pane and iframe', () => {
            const store = createMockStore([thunk])({
                error: { errors: [] },
                previewPane: {
                    page: '/',
                    pageUrl: '/?stencilEditor=789@012',
                    themePreviewConfig: {
                        configurationId: '012',
                        lastCommitId: '456',
                        versionId: '789',
                    },
                    viewportType: {
                        viewportHeight: '100%',
                        viewportWidth: '90%',
                    },
                },
                theme: {
                    configurationId: '012',
                },
            });

            const mockXMLSerializer = jest.fn(() => ({
                serializeToString: () => 'meowMeows',
            }));

            const globalAll: any = global;
            globalAll.XMLSerializer = mockXMLSerializer;

            const routes = mount(
                <Provider store={store}>
                    <StaticRouter location={styles.path} context={{}}>
                        <ThemeProvider theme={theme}>
                            <Routes {...routeProps} />
                        </ThemeProvider>
                    </StaticRouter>
                </Provider>
            );
            const previewPane = routes.find(PreviewPane);

            expect(previewPane.length).toEqual(1);
            expect(previewPane.find('iframe').prop('height')).toBe('100%');
            expect(previewPane.find('iframe').prop('width')).toBe('90%');
        });
    });

    describe('when route does not match any known path', () => {
        it('should render the 404 page', () => {
            const routes = mount(
                <StaticRouter location="/catz" context={{}}>
                    <ThemeProvider theme={theme}>
                        <Routes {...routeProps} />
                    </ThemeProvider>
                </StaticRouter>
            );
            const noRoute = routes.find(NoRoute);

            expect(noRoute.find(Heading).text()).toEqual(NoMatch.Heading);
            expect(noRoute.find(Body).text()).toEqual(NoMatch.Body);
        });
    });
});
