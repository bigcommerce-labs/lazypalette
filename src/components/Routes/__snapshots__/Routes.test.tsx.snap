// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes renders 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes
    history={[MockFunction]}
    location={[MockFunction]}
    match={[MockFunction]}
    staticContext={[MockFunction]}
  />
</Router>
`;
